cpp/src/communicator/ucxx.cpp

```diff
@@ -241,26 +239,12 @@ class SharedResources {
      * @brief Registers an endpoint without a rank.
      *
      * Registers an endpoint to a remote process with a still unknown rank.
-     * The rank must be later associated by calling the `associate_endpoint_rank()`.
      *
      * @param endpoint The endpoint to register.
      */
     void register_endpoint(std::shared_ptr<::ucxx::Endpoint> endpoint) {
         std::lock_guard<std::mutex> lock(endpoints_mutex_);
-        endpoints_[endpoint->getHandle()] = std::move(endpoint);
-    }
-
-    /**
-     * @brief Associate endpoint with a specific rank.
-     *
-     * Associate a previously registered endpoint to a specific rank.
-     *
-     * @param rank The rank to register the endpoint for.
-     * @param endpoint_handle The handle of the endpoint to register.
-     */
-    void associate_endpoint_rank(Rank const rank, ucp_ep_h const endpoint_handle) {
-        std::lock_guard<std::mutex> lock(endpoints_mutex_);
-        rank_to_endpoint_[rank] = endpoints_[endpoint_handle];
+        endpoints_[endpoint->getHandle()] = endpoint;
```

Comment:
Shall we add some overwrite protection here? 
```suggestion
       RAPIDSMPF_EXPECTS(endpoints_.emplace(endpoint->getHandle(), std::move(endpoint)), "endpoint handle already exists");
```

Reply:
same goes to L234-235. I feel like it'll make things fail fast if some things behave unexpectedly.

Reply:
Done in https://github.com/rapidsai/rapidsmpf/pull/408/commits/f02fa1881904eaf293cb50370bbbf96826cf847a